### ISSUE TITLE
Consider switching to CMake

### ISSUE THREAD

**Issue by fusepilot** [PluginCafe/cinema4d_cpp_sdk#5]
https://cmake.org/overview/

This would ( as I understand it ) allow for a few improvements.

- Maintain a single cross-platform build file instead of 2 platform specific project files ( .xcodeproj and .vcproj ). 
- Generate project files for many IDE's or build systems dynamically. ( XCode, VS, Eclipse, Make, Ninja, ...etc )
- If desired, use a single IDE (CLion) that works across platforms instead of needing to know both XCode and VS.

Bottom line, it seems like it would make maintenance easier and allow plugin developers to use their ideal toolchain to build cross-platform while not losing any current abilities.

**Comment by andreasblock** [comment]
Hi,
thanks for your suggestion.
I'll forward it to our development department.
Yet, I don't want to raise too much hope. There's quite some infrastructure based on the current build system and it is used heavily internally. This change would not be as easy as it may seem from the outside.
Bye,
Andreas
